data-prepper-plugins/cloudwatch-logs/src/main/java/org/opensearch/dataprepper/plugins/sink/client/CloudWatchLogsService.java

```diff
@@ -0,0 +1,221 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.sink.client;
+
+import io.micrometer.core.instrument.Counter;
+import org.apache.commons.lang3.time.StopWatch;
+import org.opensearch.dataprepper.metrics.PluginMetrics;
+import org.opensearch.dataprepper.model.event.Event;
+import org.opensearch.dataprepper.model.event.EventHandle;
+import org.opensearch.dataprepper.model.record.Record;
+import org.opensearch.dataprepper.plugins.sink.buffer.Buffer;
+import org.opensearch.dataprepper.plugins.sink.config.CloudWatchLogsSinkConfig;
+import org.opensearch.dataprepper.plugins.sink.exception.RetransmissionLimitException;
+import org.opensearch.dataprepper.plugins.sink.threshold.ThresholdCheck;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import software.amazon.awssdk.awscore.exception.AwsServiceException;
+import software.amazon.awssdk.core.exception.SdkClientException;
+import software.amazon.awssdk.services.cloudwatchlogs.CloudWatchLogsClient;
+import software.amazon.awssdk.services.cloudwatchlogs.model.InputLogEvent;
+import software.amazon.awssdk.services.cloudwatchlogs.model.PutLogEventsRequest;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.locks.ReentrantLock;
+
+                /*TODO: Can add DLQ logic here for sending these logs to a particular DLQ for error checking. (Explicitly for bad formatted logs).
+                    as currently the logs that are able to be published but rejected by CloudWatch Logs will simply be deleted if not deferred to
+                    a backup storage.
+                 */
+//TODO: Must also consider if the customer makes the logEvent size bigger than the send request size.
+//TODO: Can inject another class for the stopWatch functionality.
+
+public class CloudWatchLogsService {
+    public static final int LOG_EVENT_OVERHEAD_SIZE = 26; //Size of overhead for each log event message.
```

Comment:
Good question. It may make sense to make this into a computed line.

For example (and I'm just making this calculation up):

```
public static final int LOG_EVENT_OVERHEAD_SIZE = "CWLSTART[".length() + "]CWLEND";
```

If it is an approximation, then include that in the field name.

Reply:
Sorry for the lack of explanation on this one, in the official AWS documentation, there is a header that gets appended to the message, this value is held at 26. Here is the link for reference: https://docs.aws.amazon.com/AmazonCloudWatchLogs/latest/APIReference/API_PutLogEvents.html